src/kirby/directives/fit-header/fit-header.directive.ts

```diff
@@ -0,0 +1,149 @@
+import { Directive, ElementRef, Input, OnDestroy, OnInit, Renderer2 } from '@angular/core';
+
+import { ResizeObserverService } from '@kirbydesign/designsystem/components/shared/resize-observer/resize-observer.service';
+import { ResizeObserverEntry } from '@kirbydesign/designsystem/components/shared/resize-observer/types/resize-observer-entry';
+
+declare var require: any;
+
+interface Size {
+  name: string;
+  fontSize: string;
+  lineHeight: string;
+}
+
+export interface FitHeaderConfig {
+  maxLines: number;
+}
+
+@Directive({
+  selector: `h1[kirbyFitHeader],h2[kirbyFitHeader],h3[kirbyFitHeader]`,
+})
+export class FitHeaderDirective implements OnInit, OnDestroy {
+  // tslint:disable-next-line:no-input-rename
+  @Input('kirbyFitHeader') config?: FitHeaderConfig = {
+    maxLines: 2,
+  };
+
+  private lineHeight: number;
+  private height: number;
+  private width: number;
+  private clone: Element;
+  private scalingHeader: boolean; // used to prevent resizeObserver to trigger on font scaling by this.scaleHeader()
+  private scssVariables: any = require('sass-extract-loader!../../scss/base/_variables.scss');
+  private sizes: Size[] = [
+    {
+      name: 'h1',
+      fontSize: this.getFontSize('xl'),
+      lineHeight: this.getLineHeight('xl'),
+    },
+    {
+      name: 'h2',
+      fontSize: this.getFontSize('l'),
+      lineHeight: this.getLineHeight('l'),
+    },
+    {
+      name: 'h3',
+      fontSize: this.getFontSize('m'),
+      lineHeight: this.getLineHeight('m'),
+    },
+  ];
+
+  constructor(
+    private elementRef: ElementRef,
+    private renderer: Renderer2,
+    private resizeObserverService: ResizeObserverService
+  ) {}
+
+  ngOnInit(): void {
+    this.observeResize();
+  }
+
+  ngOnDestroy(): void {
+    this.resizeObserverService.unobserve(this.elementRef);
+    this.renderer.removeChild(this.elementRef.nativeElement, this.clone);
+  }
+
+  private getFontSize(size): string {
+    if (
```

Comment:
Consider simplyfing with helper method:
```
private getScssValue(path: string[]) {
    let node = this.scssVariables;
    for (let step of path) {
      node = node[step];
      if (!node) {
        break;
      }
    }
    return node;
  }
```
and use like:
`const fontSize = this.getScssValue(['global', '$font-sizes', 'value', size]);`